src/decoder/mod.rs

```diff
@@ -1066,54 +1164,98 @@ impl<R: Read + Seek> Decoder<R> {
         self.read_strip_to_buffer(result.as_buffer(0))?;
 
-        Ok(result)
+        Ok((result, chunk_info))
     }
 
-    /// Read a single tile from the image and return it as a Vector
+    /// Read a single tile from the image and return it as a Vector. This method does not return
+    /// information on the size of the tile read and so the caller must take care of tiles on the
+    /// rightmost column and the bottommost row in the image as the data read may be less than the
+    /// expected tile size. This interface may change at a future date to reflect this.
+    ///
+    /// `read_tile_at` can be used to get both the tile data and the tile size.
+    #[deprecated = "Use read_chunk instead"]
     pub fn read_tile(&mut self) -> TiffResult<DecodingResult> {
-        self.check_chunk_type(ChunkType::Tile)?;
+        let result = self.read_tile_at(self.current_chunk)?;
 
-        let tile = self.current_chunk;
+        self.current_chunk += 1;
 
-        let tile_attrs = self.image().tile_attributes.as_ref().unwrap();
-        let (padding_right, padding_down) = tile_attrs.get_padding(tile);
+        Ok(result.0)
+    }
 
-        let tile_width = tile_attrs.tile_width - padding_right;
-        let tile_length = tile_attrs.tile_length - padding_down;
+    /// Read the specified tile (at index `tile_index`) and return the binary data as a Vector and
+    /// `ChunkInfo` describing the size of the tile.
+    #[deprecated = "Use read_chunk_at instead"]
+    pub fn read_tile_at(&mut self, tile_index: usize) -> TiffResult<(DecodingResult, ChunkInfo)> {
+        self.check_chunk_type(ChunkType::Tile)?;
 
-        let mut result = self.result_buffer(tile_width, tile_length)?;
+        let chunk_info = self
+            .image()
+            .chunk_info(tile_index)
+            .ok_or(TiffFormatError::InvalidChunkIndex(tile_index))?;
 
-        let offset = self.image.chunk_file_range(tile)?.0;
-        self.goto_offset_u64(offset)?;
+        let mut result = self.result_buffer(chunk_info.data_width, chunk_info.data_height)?;
 
-        let byte_order = self.reader.byte_order;
-        self.image.expand_chunk(
-            &mut self.reader,
-            result.as_buffer(0),
-            tile_width,
-            byte_order,
-            tile,
-        )?;
+        self.read_tile_to_buffer(result.as_buffer(0), tile_index, chunk_info.data_width)?;
+
+        Ok((result, chunk_info))
+    }
+
+    /// Read a single chunk from the image and return it as a Vector. This method does not return
+    /// information on the size of the chunk read and so the caller must take care of chunks on the
+    /// rightmost column and the bottommost row in the image as the data read may be less than the
+    /// expected chunk size. This interface may change at a future date to reflect this.
+    ///
+    /// `read_chunk_at` can be used to get both the chunk data and the tile size.
+    pub fn read_chunk(&mut self) -> TiffResult<DecodingResult> {
```

Comment:
Do we need this method? We could recommend always using `read_chunk_at` instead.